tt_metal/impl/device/device_pool.cpp

```diff
@@ -33,55 +33,86 @@ std::unordered_map<int, std::vector<uint32_t>> get_cpu_cores_per_numa_node(std::
     return cpu_cores_per_numa_node;
 }
 
-int get_cpu_core_for_device_worker_thread(
+std::pair<int, int> get_cpu_core_for_device_worker_thread(
     int mmio_controlled_device_id,
     const std::unordered_map<int, std::vector<uint32_t>>& cpu_cores_per_numa_node,
-    std::unordered_set<uint32_t>& free_cores) {
-    int core_assigned_to_device = 0;
-    if (numa_available() != -1) {
-        // Get NUMA node that the current device is mapped to through UMD
-        int numa_node_for_device = tt::Cluster::instance().get_numa_node_for_device(mmio_controlled_device_id);
-        if (cpu_cores_per_numa_node.find(numa_node_for_device) != cpu_cores_per_numa_node.end()) {
-            // NUMA node reported by UMD exists on host. Choose a core on this numa-node using round robin policy
-            int num_cores_in_numa_node = cpu_cores_per_numa_node.at(numa_node_for_device).size();
-            core_assigned_to_device =
-                cpu_cores_per_numa_node.at(numa_node_for_device).at(mmio_controlled_device_id % num_cores_in_numa_node);
+    std::unordered_set<uint32_t>& free_cores,
+    uint32_t num_devices,
+    bool use_separate_procs) {
+
+    int core_assigned_to_device_worker_thread = 0;
+    int core_assigned_to_device_completion_queue_reader = 0;
+    uint32_t num_online_processors = sysconf(_SC_NPROCESSORS_ONLN);
+    // Get NUMA node that the current device is mapped to through UMD
+    int numa_node_for_device = tt::Cluster::instance().get_numa_node_for_device(mmio_controlled_device_id);
+
+    if (numa_available() != -1 and cpu_cores_per_numa_node.find(numa_node_for_device) != cpu_cores_per_numa_node.end()) {
+        // NUMA node reported by UMD exists on host. Choose a core on this numa-node using round robin policy
+        int num_cores_in_numa_node = cpu_cores_per_numa_node.at(numa_node_for_device).size();
```

Comment:
a couple of access to `cpu_cores_per_numa_node.at(numa_node_for_device)` for readability could you pull this out?

Reply:
Done